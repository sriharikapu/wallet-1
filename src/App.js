import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import ethers from 'ethers';
import CryptoJS from 'crypto-js';

class App extends Component {
  state = {
    username: '',
    walletPassword: ''
  };

  handleSubmit = async (username, walletPassword) => {
    try {
      let wallet = ethers.Wallet.createRandom();
      let jsonWallet = await wallet.encrypt(walletPassword, {});
      let backendPassword = CryptoJS.HmacSHA256(
        username,
        walletPassword
      ).toString();

      // sessionStorage['username'] = username;
      // sessionStorage['jsonWallet'] = jsonWallet;

      console.log('jsonWallet', jsonWallet);
    } catch (err) {
      console.log('err', err);
    }
  };

  handleChange = (field, value) => this.setState({ [field]: value });

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
        <form
          onSubmit={e => {
            e.preventDefault();
            this.handleSubmit(this.state.username, this.state.walletPassword);
          }}
        >
          <input
            type="text"
            placeholder="username"
            onChange={e => this.handleChange('username', e.target.value)}
          />
          <input
            type="text"
            placeholder="password"
            onChange={e => this.handleChange('walletPassword', e.target.value)}
          />
          <input type="submit" />
        </form>
      </div>
    );
  }
}

export default App;
