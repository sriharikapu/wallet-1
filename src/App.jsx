import React, { PureComponent, Fragment } from 'react';
import Login from './features/authentication/Login';
import Register from './features/authentication/Register';
import RetrieveWithPrivateKey from './features/authentication/Retrieve';
import Navbar from './components/Nav';
import styled from 'styled-components';
import getWeb3 from './constants/web3';
import Tx from './features/transactions/Tx';
import { Button } from 'antd';

const LoggedIn = ({ logout, web3, title }) => (
  <Fragment>
    <br />
    <br />
    <Button type="danger" size="small" onClick={logout}>
      Logout
    </Button>
    <br />
    <br />
    <br />
    <br />
    <Tx web3={web3} title={title} />
  </Fragment>
);

const LoggedOut = ({ updateTitle, web3, setUser }) => (
  <Fragment>
    <Register updateTitle={updateTitle} web3={web3} />
    <br />
    <br />
    <Login updateTitle={updateTitle} setUser={setUser} web3={web3} />
    <br />
    <br />
    <RetrieveWithPrivateKey updateTitle={updateTitle} web3={web3} />
  </Fragment>
);

const Wrapper = styled.div`
  text-align: center;
`;

class App extends PureComponent {
  state = {
    title: 'Welcome to Usabl',
    auth: false,
    balance: '',
    web3: null
  };

  componentWillMount() {
    getWeb3
      .then(results => {
        this.setState({
          web3: results.web3
        });
        // Instantiate contract once web3 provided.
        // this.instantiateContract();
      })
      .catch(() => {
        console.log('Error finding web3.');
      });
  }

  updateTitle = async jsonWallet => {
    let balance = await this.state.web3.eth.getBalance(jsonWallet.address);
    this.setState(() => ({
      title: `0x${jsonWallet.address}`,
      auth: true,
      balance
    }));
  };

  setUser = async user => {
    let balance = await this.state.web3.eth.getBalance(user.jsonWallet.address);
    this.setState(() => ({
      balance,
      title: `0x${user.jsonWallet.address}`,
      auth: true
    }));
  };

  logout = () =>
    this.setState(() => ({
      title: 'Welcome to Usabl',
      auth: false,
      balance: ''
    }));

  render() {
    let { auth, title, balance } = this.state;
    return (
      <Wrapper>
        <Navbar title={title} balance={balance} />
        {!auth ? (
          <LoggedOut
            updateTitle={this.updateTitle}
            web3={this.state.web3}
            setUser={this.setUser}
          />
        ) : (
          <LoggedIn
            logout={this.logout}
            web3={this.state.web3}
            title={this.state.title}
          />
        )}
      </Wrapper>
    );
  }
}

export default App;
