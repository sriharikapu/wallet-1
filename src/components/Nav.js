import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

class Navbar extends PureComponent {
  static propTypes = { title: PropTypes.string.isRequired };

  render() {
    let { title } = this.props;
    return (
      <header className="App-header">
        <h1 className="App-title">{title}</h1>
      </header>
    );
  }
}

export default Navbar;
