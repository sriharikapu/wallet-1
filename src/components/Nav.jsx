import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Wrapper = styled.div`
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
`;

class Navbar extends PureComponent {
  static propTypes = { title: PropTypes.string.isRequired };

  render() {
    let { title, balance } = this.props;
    return (
      <Wrapper>
        <h1>{title}</h1>
        <h1>{balance}</h1>
      </Wrapper>
    );
  }
}

export default Navbar;
