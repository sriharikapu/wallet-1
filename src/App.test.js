import React from 'react';
import { render, cleanup } from 'react-testing-library';
import 'jest-dom/extend-expect';
import 'jest-styled-components';
import App from './App';
import Nav from './components/Nav';
import TxModal from './features/transactions/Tx';

afterEach(cleanup);

test('renders without crashing', () => {
  const { getByTestId } = render(<App />);
  expect(getByTestId('mainApp')).toBeTruthy();
});

test('renders Nav with props', () => {
  const { getByTestId } = render(<Nav title="test title" balance={2} />);
  expect(getByTestId('navbar')).toHaveTextContent('test title');
  expect(getByTestId('navbar')).toHaveTextContent(2);
});

test('nav matches snapshot', () => {
  const { getByTestId } = render(<Nav title="test title" balance={2} />);
  expect(getByTestId('navbar')).toMatchSnapshot();
});

test.skip('TX modal matches snapshot', () => {
  const mockFn = jest.fn();
  const { getByTestId } = render(
    <TxModal
      visible={false}
      handleOk={mockFn}
      handleCancel={mockFn}
      handleChange={mockFn}
      passwordConfirm={'test password'}
      gas={100}
      showAdvanced={false}
      onChange={mockFn}
      closeModal={mockFn}
      loading={false}
      min={10}
      max={100}
      actualPrice={100}
    />,
  );
  expect(getByTestId('txModal')).toMatchSnapshot();
});
