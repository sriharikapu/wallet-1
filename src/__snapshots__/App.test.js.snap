// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TX modal matches snapshot 1`] = `
<nav
  class="sc-bxivhb cEOKjk"
  data-testid="navbar"
>
  <h1>
    Your address is
  </h1>
  <h2
    class="sc-ifAKCX gmEbVI"
  >
    test title
  </h2>
  <div>
    <div
      class="sc-EHOje hXnmrd"
    >
      2
    </div>
  </div>
</nav>
`;

exports[`nav matches snapshot 1`] = `
<nav
  class="sc-bxivhb cEOKjk"
  data-testid="navbar"
>
  <h2
    class="sc-ifAKCX gmEbVI"
  >
    test title
  </h2>
  <div>
    <div
      class="sc-EHOje hXnmrd"
    >
      2
    </div>
  </div>
</nav>
`;
